distro/rpm/openvpn.init.d.rhel

```diff
@@ -224,17 +224,24 @@ case "$1" in
 	fi
 	;;
   status)
-	if [ -f $lock ]; then
-	    for pidf in `/bin/ls $piddir/*.pid 2>/dev/null`; do
-		if [ -s $pidf ]; then
-		    kill -USR2 `cat $pidf` >/dev/null 2>&1
-		fi
-	    done
-	    echo "Status written to /var/log/messages"
-	else
-	    echo "openvpn: service not started"
-	    exit 1
-	fi
+        if [ -f $lock ]; then
+            for pidf in `/bin/ls $piddir/*.pid 2>/dev/null`; do
+                PID=$(cat $pidf)
+                if [ -e /proc/$PID ]; then
+                    echo "openvpn is running. pid: $PID"
+                    exit 0
+                else
+                    echo "openvpn is dead but lock file exists: $lock and pid file exists: $pidf"
+                    exit 1
+                fi
+            done 
```

Comment:
I understand your concern. But at the same time I hardly can imagine use
case when several openvpn instances are using same directory to store pid
files. From my experience,
such instances would be configured to use different service names and
different values for $piddir and few other variables for example by
creating a separate init script file to control each instance.

On Aug 16, 2016 02:05, "Samuli Seppänen" <notifications@github.com> wrote:

> In distro/rpm/openvpn.init.d.rhel
> <https://github.com/OpenVPN/openvpn/pull/59#discussion_r74899640>:
>
> > -	    echo "Status written to /var/log/messages"
> > -	else
> > -	    echo "openvpn: service not started"
> > -	    exit 1
> > -	fi
> > +        if [ -f $lock ]; then
> > +            for pidf in `/bin/ls $piddir/*.pid 2>/dev/null`; do
> > +                PID=$(cat $pidf)
> > +                if [ -e /proc/$PID ]; then
> > +                    echo "openvpn is running. pid: $PID"
> > +                    exit 0
> > +                else
> > +                    echo "openvpn is dead but lock file exists: $lock and pid file exists: $pidf"
> > +                    exit 1
> > +                fi
> > +            done
>
> So this part loops through all pidfiles in $piddir, but exits with 0 or 1,
> depending on the status of the first OpenVPN process /bin/ls returns. So,
> if there are several OpenVPN instances running, the return value of the
> "service status" depends on the status of the first OpenVPN instance, which
> I think is suboptimal.
>
> That said, I don't know if that is the best we can do, or what is the
> convention in these cases. Thoughts?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenVPN/openvpn/pull/59/files/fa320fb111d808fdcc39ee6d2870265b4bc8790c#r74899640>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF4L_ROCp20DeHWObnkWTM8Sd70vnpRSks5qgX11gaJpZM4JTAOi>
> .
>